File: Prefix & Suffix Sum/1685. Sum of Absolute Differences in a Sorted Array.java
class Solution {
    public int[] getSumAbsoluteDifferences(int[] nums) {
        int n = nums.length;
        int[] pri = new int[n];
        pri[0] = nums[0];
        for (int i = 1; i < n; i++) {
            pri[i] = nums[i] + pri[i - 1];

        }
        int[] ans = new int[n];
        for (int i = 0; i < n; i++) {
            int leftsum = pri[i] - nums[i];
            int rightsum = pri[n - 1] - pri[i];
            int leftc = i;
            int rightc = n - 1 - i;
            int leftans = nums[i] * leftc - leftsum;
            int rightans = rightsum - rightc * nums[i];
            ans[i] = leftans + rightans;

        }
        return ans;

    }
}

File: LeetCode/Prefix & Suffix Sum Technique/1685_Sum of Absolute Differences in a Sorted Array.py
class Solution:
    def getSumAbsoluteDifferences(self, nums: List[int]) -> List[int]:
        n = len(nums)
        prefix_sum = [0] * (n + 1)
        for i in range(n):
            prefix_sum[i + 1] = prefix_sum[i] + nums[i]

        result = [0] * n
        for i in range(n):
            left_sum = nums[i] * i - prefix_sum[i]
            right_sum = (prefix_sum[n] - prefix_sum[i + 1]) - nums[i] * (n - i - 1)
            result[i] = left_sum + right_sum
        
        return result

user_input = input("Enter a list of integers separated by spaces: ")
nums = list(map(int, user_input.split()))
solution = Solution()
output = solution.getSumAbsoluteDifferences(nums)
print(f"The sum of absolute differences for the given list is: {output}")
